Support blank curlies

gjtorikian opened the pull request:
`print_schema` will happily print empty curlies for types without fields `type Foo{}`, but the parser will not accept this as a valid IDL file:

This PR changes parser.y to accept empty field lists. This is also in compliance with the IDL spec at https://github.com/facebook/graphql/pull/90/files#r132484441.

xref https://github.com/gjtorikian/graphql-docs/issues/32
cc @Mori

rmosolgo commented:
This looks great, thanks for taking care of it! 

But, I see that some changes were committed to `lexer.rb`, was this intentional? That file is generated by `ragel`, based on `lexer.rl`, so I don't expect any changes to it, except when the `.rl` file also changes. Do you mind reverting the changes to `lexer.rb`?

gjtorikian commented:
Hmm, I sample ran the rake task but sure I’ll try reverting that change!

gjtorikian commented:
`rake build_parser` insists on including lexer.rb:

```
graphql-ruby (support-blank-curlies): git status          
On branch support-blank-curlies
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	modified:   lib/graphql/language/parser.rb
	modified:   lib/graphql/language/parser.y

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   lib/graphql/language/lexer.rb

graphql-ruby (support-blank-curlies): git checkout lib/   
graphql-ruby (support-blank-curlies): git status       
On branch support-blank-curlies
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	modified:   lib/graphql/language/parser.rb
	modified:   lib/graphql/language/parser.y

graphql-ruby (support-blank-curlies): be rake build_parser
135 shift/reduce conflicts
45 reduce/reduce conflicts
graphql-ruby (support-blank-curlies): git status          
On branch support-blank-curlies
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	modified:   lib/graphql/language/parser.rb
	modified:   lib/graphql/language/parser.y

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   lib/graphql/language/lexer.rb

```

Have I done something wrong?

rmosolgo commented:
Very weird, the commands all look right! I assumed it was some kind of format-on-save plugin, but I guess not 😖 

Anyhow, I pulled the branch and re-ran `build_parser`. It cleared up the changes to `lexer.rb`, but I was surprised to also see [changes to `parser.rb`](https://github.com/rmosolgo/graphql-ruby/pull/1145/commits/619dabeb785fb20a089bee5b6ba0e3e5967b863a#diff-85a436d444a3faed35353741f5108507). Then, I realized that I don't actually _check_ the Racc & Ragel versions on people's dev environment when they build those files, so it could be that different versions output slightly different code. So, I added those checks to make sure we're all running the same versions when building those files. Maybe that was it, hopefully that will catch it in the future!

gjtorikian commented:
Woop woop, thank you!

rmosolgo commented:
:metal: !

zzak commented on Rakefile:
```diff
@@ -40,6 +40,26 @@ end
+    end
+  end
+
+  assert_dependency_version("Ragel", "7.0.0.9", "ragel -v")
```
@rmosolgo @gjtorikian Is there a reason to pin to this particular version of ragel?

I'm only able to find 6.9 in my [apt repos](https://packages.ubuntu.com/artful/ragel).

rmosolgo replied:
It was found to be significantly faster back in https://github.com/rmosolgo/graphql-ruby/pull/737, but I didn't add the lock until here. Yeah, sorry, you have to get the "Development" build from http://www.colm.net/open-source/ragel/!
